### PR TITLE
Remove duplicate and unnecessary use statements

### PR DESCRIPTION
**Pull request by rmontuoro**
## Description

This PR removes duplicate and unnecessary `use` statements.

## Motivation and Context

Code maintenance and cleanup
## How Has This Been Tested?

These changes were tested on Orion, building MAPL as standalone library and using it to build and run UFS-GOCART.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (affects only documentation or cleanup)

## Checklist:

- [ ] I have tested this change with a run of GEOSgcm (if non-trivial)
- [x] I have added one of the required labels (0 diff, 0 diff trivial, 0 diff structural, non 0-diff)
- [ ] I have updated the CHANGELOG.md accordingly following the style of [Keep a Changelog](https://keepachangelog.com/en/1.0.0/#how)

### CODE REVIEW

File: generic/MaplGeneric.F90 [hunk 1]
```diff
@@ -2,7 +2,6 @@ module MaplGeneric
    use mapl_AbstractComponent
    use mapl_MaplComponent
    use mapl_ComponentSpecification
-   use mapl_VariableSpecification
```
Comment (tclune) [inline generic/MaplGeneric.F90:5]: Is this change necessary?   The other layers of MAPL don't yet use this (new) data type, but they will eventually.

File: pfio/MultiCommServer.F90 [hunk 2]
```diff
@@ -17,7 +17,6 @@ module pFIO_MultiCommServerMod
    use pFIO_ServerThreadVectorMod
    use pFIO_AbstractSocketMod
    use pFIO_AbstractSocketVectorMod
-   use pFIO_AbstractDataReferenceMod
```
Comment (tclune) [inline pfio/MultiCommServer.F90:20]: Just a question on this one:    are you just noticing that some of these USE's are unnecessary?    How do you detect this.

**Review by tclune — COMMENTED** [review]
Will approve module response to the first question.

File: generic/MaplGeneric.F90 (hunk 1 above)
Comment (rmontuoro) [reply generic/MaplGeneric.F90:5]: @tclune This change attempts to fix the following compiler error:
```
[...]/GOCART/ESMF/GOCART2G_GridComp/CA2G_GridComp/CA2G_GridCompMod.F90(894): error #6405: The same named entity from different modules and/or program units cannot be referenced.   [MAPL_VARSPEC]
    type(MAPL_VarSpec), pointer       :: InternalSpec(:)
```

Module `mapl_VariableSpecification` exposes derived type `MAPL_VarSpec`, which is also exposed by module `MAPL_VarSpecMod`. Based on last week's discussions, my understanding is that `MAPL_VarSpecMod` is the proper "owner" of `MAPL_VarSpec`.

**Comment by mathomp4** [comment]
Actually, this looks to be against `main`. This should be a PR to `develop`

File: pfio/MultiCommServer.F90 (hunk 2 above)
Comment (rmontuoro) [reply pfio/MultiCommServer.F90:20]: @tclune This `use` statement is duplicated. See the [identical statement](https://github.com/rmontuoro/MAPL/blob/beb095f4a14f008fbbef061a135b31a1dab12236/pfio/MultiCommServer.F90#L23) still in the source code (line 23).

**Comment by rmontuoro** [comment]
> Actually, this looks to be against `main`. This should be a PR to `develop`

This PR fixes a compiler error (see comment above), that's why it was issued against `main`.

**Comment by mathomp4** [comment]
> > Actually, this looks to be against `main`. This should be a PR to `develop`
> 
> This PR fixes a compiler error (see comment above), that's why it was issued against `main`.

Ah. GitHub decided not to refresh my page. Thanks web caching!

**Comment by tclune** [comment]
This PR should also be for develop.  Sorry for the extra work.

**Comment by rmontuoro** [comment]
Updated and issued against `develop`.

File: generic/MaplGeneric.F90 (hunk 1 above)
Comment (tclune) [reply generic/MaplGeneric.F90:5]: I thought the solution we discussed was a change in MAPL_Generic.F90.   `MAPL/generic` is the proper owner of this module and should be exporting this.

The error you refer to should not happen when starting a clean build.    If it does, lets handle this as a separate hotfix on main.

(These changes otherwise all need to be redone from a base off of develop.  Sorry)
